es6-shim.js

```diff
@@ -1241,29 +1241,29 @@
   var MathShims = {
     acosh: function acosh(value) {
       var x = Number(value);
-      if (Number.isNaN(x) || value < 1) { return NaN; }
-      if (x === 1) { return 0; }
-      if (x === Infinity) { return x; }
-      return Math.log(x / Math.E + Math.sqrt(x + 1) * Math.sqrt(x - 1) / Math.E) + 1;
+      if (x * x + 1 === x * x) {
+        return Math.log(x) + Math.log(2);
+      }
+      return Math.log1p(x - 1 + Math.sqrt(x - 1) * Math.sqrt(x + 1));
     },
 
     asinh: function asinh(value) {
       var x = Number(value);
-      if (x === 0 || !globalIsFinite(x)) {
+      if (x === 0) {
         return x;
       }
-      return x < 0 ? -Math.asinh(-x) : Math.log(x + Math.sqrt(x * x + 1));
+      if (x < 0) {
+        return -Math.asinh(-x);
+      }
+      if (x * x + 1 === x * x) {
+        return Math.log(x) + Math.log(2);
+      }
+      return Math.log1p(x + Math.expm1(Math.log1p(x * x) / 2));
```

Comment:
@lewisje , 
Hi! Why is it better? Could you provide more details?
Your implementation is not accurate for small values (Math.pow(2, -1021)), 

What I do not like in my is : `if (x * x + 1 === x * x) {`. Possibly, I should use `x > Math.sqrt(2 / Number.EPSILON)` here.